addons/fetchmail/models/fetchmail.py

```diff
@@ -17,6 +18,10 @@
 # Workaround for Python 2.7.8 bug https://bugs.python.org/issue23906
 poplib._MAXLINE = 65536
 
+# Add timeout to IMAP connections
+# HACK https://bugs.python.org/issue38615
+IMAP4._create_socket = lambda self: socket.create_connection((self.host or None, self.port), MAIL_TIMEOUT)
```

Comment:
what about `IMAP4_SSL` ?

Reply:
IMAP4_SSL is a subclass of IMAP4 and wraps the socket created by IMAP4._create_socket, so it works for both normally:
https://github.com/python/cpython/blob/3.9/Lib/imaplib.py#L1326